xrdsst/controllers/security_server.py

```diff
@@ -0,0 +1,150 @@
+        stacked_on = 'base'
+        stacked_type = 'nested'
+        description = texts['security_server.controller.description']
+
+    @ex(help="List security servers", arguments=[])
+    def list(self):
+        active_config = self.load_config()
+        self.list_security_servers(active_config)
+
+    @ex(help="List security server versions", arguments=[])
+    def version(self):
+        active_config = self.load_config()
+        self.list_security_servers_version(active_config)
+
+    def list_security_servers(self, config):
+        ss_api_conf_tuple = list(zip(config["security_server"], map(lambda ss: self.create_api_config(ss, config), config["security_server"])))
+        render_data = []
+        render_list = []
+        for security_server in config["security_server"]:
+            ss_api_config = self.create_api_config(security_server, config)
+            ss_list = self.remote_list_security_servers(ss_api_config)
+            if ss_list:
+                render_list = self.union_security_server_lists(render_list, ss_list)
+        if len(render_list) > 0:
+            if self.is_output_tabulated():
+                render_data = [SecurityServerListMapper.headers()]
+                render_data.extend(map(SecurityServerListMapper.as_list, render_list))
+            else:
+                render_data.extend(map(SecurityServerListMapper.as_object, render_list))
+            self.render(render_data)
+        BaseController.log_keyless_servers(ss_api_conf_tuple)
+        return render_list
+
+    def list_security_servers_version(self, config):
+        ss_api_conf_tuple = list(zip(config["security_server"], map(lambda ss: self.create_api_config(ss, config), config["security_server"])))
+        render_data = []
+        render_list = []
+        for security_server in config["security_server"]:
+            ss_api_config = self.create_api_config(security_server, config)
+            version = self.remote_list_security_server_version(ss_api_config)
+            if version:
+                render_list.append({
+                    'security_server': security_server["name"],
+                    'version': version
+                })
+        if len(render_list) > 0:
+            if self.is_output_tabulated():
+                render_data = [SecurityServerVersionMapper.headers()]
+                render_data.extend(map(SecurityServerVersionMapper.as_list, render_list))
+            else:
+                render_data.extend(map(SecurityServerVersionMapper.as_object, render_list))
+            self.render(render_data)
+        BaseController.log_keyless_servers(ss_api_conf_tuple)
+        return render_list
+
+    def remote_list_security_servers(self, ss_api_config):
+        security_servers_api = SecurityServersApi(ApiClient(ss_api_config))
+        try:
+            security_servers = self.parse_security_server_response(security_servers_api.get_security_servers())
+            return list(security_servers)
```

Comment:
get_security_servers() already returns a list, so you are creating a list of lists here?

Reply:
no, i doesn't return a list, returns other type of generated object, it works fine with the cast to list